GlobalSettings.php

```diff
@@ -36,6 +36,35 @@
 	};
 }
 
+// log suspicious or sensitive login attempts
+$wgHooks['AuthManagerLoginAuthenticateAudit'][] = static function ( $response, $user, $username ) {
+	$guessed = false;
+	if ( !$user && $username ) {
+		$user = MediaWikiServices::getInstance()
+			->getUserFactory()
+			->newFromName( $username );
+		$guessed = true;
+	}
+	if ( !$user || !in_array( $response->status,
+		[ AuthenticationResponse::PASS, AuthenticationResponse::FAIL ], true )
+	) {
+		return;
+	}
+	global $wgRequest;
```

Comment:
We shouldn't use global $wgRequest

Reply:
Alternative?

Reply:
`RequestContext::getMain()->getRequest()` with a `use MediaWiki\Context\RequestContext;` also